File: C++/Program/Handphone.cpp
#include <iostream>
#include <string>
#include "BarangElektronik.cpp"

using  namespace std;

class Handphone : public BarangElektronik{
    private:
    string merek;
    string warna;
    int ram;
    int storage;


    public:
    Handphone(){

    }

    //contruktor

    Handphone(int kodeProduk, int stok, int harga, string namaBarang, int garansi, int tahunRilis, string merek, string warna,
             int ram, int storage) : BarangElektronik(kodeProduk, stok, harga, namaBarang, garansi, tahunRilis) {
        this->merek = merek;
        this->warna = warna;
        this->ram = ram;
        this->storage = storage;


    }

    //getter dan setter

     string getMerek(){
           return merek;
        }
        
     void setMerek(string merek){
           this->merek = merek;
        }

     string getWarna(){
            return warna;
        }

     void setWarna(string warna){
          this-> warna= warna;
        }

     int getRam(){
          return ram;
        }

     void setRam(int ram){
          this->ram = ram;
        }

      int getStorage(){
          return storage;
        }

     void setStorage(int storage){
          this->storage = storage;
        }













};

File: php/Program/Handphone.php
<?php

class Handphone extends BarangElektronik {
    private $merek;
    private $warna;
    private $ram;
    private $storage;

    public function __construct($kodeProduk, $stok, $harga, $namaBarang, $garansi, $tahunRilis,
                                $merek, $warna, $ram, $storage) {
        parent::__construct($kodeProduk, $stok, $harga, $namaBarang, $garansi, $tahunRilis);
        $this->merek = $merek;
        $this->warna = $warna;
        $this->ram = $ram;
        $this->storage = $storage;
    }

   ///getter dan setter
    public function getMerek() {
        return $this->merek;
    }
    public function setMerek($merek) {
        $this->merek = $merek;
    }

    public function getWarna() {
        return $this->warna;
    }
    public function setWarna($warna) {
        $this->warna = $warna;
    }

    public function getRam() {
        return $this->ram;
    }
    public function setRam($ram) {
        $this->ram = $ram;
    }
    public function getStorage() {
        return $this->storage;
    }
    public function setStorage($storage) {
        $this->storage = $storage;
    }

}

